gateway/src/apicast/cli/environment.lua

```diff
@@ -0,0 +1,155 @@
+--- Environment configuration
+-- @module environment
+-- This module is providing a configuration to APIcast before and during its initialization.
+-- You can load several configuration files.
+-- Fields from the ones added later override fields from the previous configurations.
+local pl_path = require('pl.path')
+local resty_env = require('resty.env')
+local linked_list = require('apicast.linked_list')
+local setmetatable = setmetatable
+local loadfile = loadfile
+local pcall = pcall
+local require = require
+local assert = assert
+local error = error
+local print = print
+local pairs = pairs
+local ipairs = ipairs
+local tostring = tostring
+local insert = table.insert
+local concat = table.concat
+local re = require('ngx.re')
+
+local function parse_nameservers()
+    local resolver = require('resty.resolver')
+    local nameservers = {}
+
+    for _,nameserver in ipairs(resolver.init_nameservers()) do
+        insert(nameservers, tostring(nameserver))
+    end
+
+    if #nameservers > 0 then
+        return nameservers
+    end
+end
+
+
+local _M = {}
+---
+-- @field default_environment Default environment name.
+-- @table self
```

Comment:
Couldn't figure out a better way to document this. This actually means there would be a property `self` of the module returned by this file. 
But there is no way I could find to have properties of the module just plain values with description. Only tables can be properties of the module and only tables can have fields.